span.go

```diff
@@ -6,10 +6,54 @@ import (
 	"time"
 )
 
+// SpanContext represents Span state that should propagate to descendant Spans
+// and presumably across process boundaries.
+type SpanContext interface {
+	// SetBaggageItem sets a key:value pair on this Span that also
+	// propagates to future Span children.
+	//
+	// SetBaggageItem() enables powerful functionality given a full-stack
+	// opentracing integration (e.g., arbitrary application data from a mobile
+	// app can make it, transparently, all the way into the depths of a storage
+	// system), and with it some powerful costs: use this feature with care.
+	//
+	// IMPORTANT NOTE #1: SetBaggageItem() will only propagate trace
+	// baggage items to *future* children of the Span.
+	//
+	// IMPORTANT NOTE #2: Use this thoughtfully and with care. Every key and
+	// value is copied into every local *and remote* child of this Span, and
+	// that can add up to a lot of network and cpu overhead.
+	//
+	// IMPORTANT NOTE #3: Baggage item keys have a restricted format:
+	// implementations may wish to use them as HTTP header keys (or key
+	// suffixes), and of course HTTP headers are case insensitive.
+	//
+	// As such, `restrictedKey` MUST match the regular expression
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
+	// start with a letter or number, and the remaining characters must be
+	// letters, numbers, or hyphens. See CanonicalizeBaggageKey(). If
+	// `restrictedKey` does not meet these criteria, SetBaggageItem()
+	// results in undefined behavior.
+	//
+	// Returns a reference to this Span for chaining, etc.
+	SetBaggageItem(restrictedKey, value string) SpanContext
+
+	// Gets the value for a baggage item given its key. Returns the empty string
+	// if the value isn't found in this Span.
+	//
+	// See the `SetBaggageItem` notes about `restrictedKey`.
+	BaggageItem(restrictedKey string) string
+}
+
 // Span represents an active, un-finished span in the OpenTracing system.
 //
 // Spans are created by the Tracer interface.
 type Span interface {
+	// Returns the SpanContext for this Span. Note that the return value of
+	// SpanContext() is still valid after a call to Span.Finish(), as is a call
+	// to Span.SpanContext() after a call to Span.Finish().
+	SpanContext() SpanContext
```

Comment:
I wonder why not embed SpanContext in Span, so that a Span can be passed as an argument to functions accepting SpanContext?